library/core/src/main/java/com/google/android/exoplayer2/video/MediaCodecVideoRenderer.java

```diff
@@ -462,9 +498,24 @@ private static List<MediaCodecInfo> getDecoderInfos(
     if (mimeType == null) {
       return ImmutableList.of();
     }
-    List<MediaCodecInfo> decoderInfos =
-        mediaCodecSelector.getDecoderInfos(
-            mimeType, requiresSecureDecoder, requiresTunnelingDecoder);
+
+    List<MediaCodecInfo> decoderInfos = new ArrayList<MediaCodecInfo>();
+    if (Util.SDK_INT >= 24) {
+      /*
+        platform with API level higher than 24 will consider following keys:
+        MediaFormat#KEY_PROFILE,
+        MediaFormat#KEY_LEVEL,
+        MediaFormat#KEY_BIT_RATE,
+       */
+      decoderInfos.addAll(
+          tryEvaluatingCodecByFormat(format, requiresSecureDecoder, requiresTunnelingDecoder));
+    }
```

Comment:
This change modifies how codecs selected, and it would apply for every app. The change also of negates the reason why MediaCodecSelector exists, since an app might inject a custom MediaCodecSelector but the MediaCodecVideoRenderer would ignore it.

I think a more suitable solution would be to change the MediaCodecSelector to receive a Format instead of the MIME type as a String, and then it will be possible for apps who wish to use MediaCodecList.findDecoderForFormat() to inject their own MediaCodecSelector.

That said, changing the MediaCodecSelector API is a wider change and it might be more efficient for the ExoPlayer team to do it internally. Please see releated comment on the conversation section